### PR TITLE
drop deprecated @platformName and @platformStatic annotations

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: compiler/backend/src/org/jetbrains/kotlin/codegen/FunctionCodegen.java [hunk 1]
```diff
@@ -209,7 +209,7 @@ public void generateMethod(
             generateMethodBody(mv, functionDescriptor, methodContext, jvmSignature, strategy, memberCodegen);
         }
         else if (staticInCompanionObject) {
-            // native platformStatic foo() in companion object should delegate to the static native function moved to the outer class
+            // native @JvmStatic foo() in companion object should delegate to the static native function moved to the outer class
             mv.visitCode();
             FunctionDescriptor staticFunctionDescriptor = PlatformStaticGenerator.createStaticFunctionDescriptor(functionDescriptor);
```
Comment (udalov) [inline compiler/backend/src/org/jetbrains/kotlin/codegen/FunctionCodegen.java]: rename `PlatformStaticGenerator` -> `JvmStaticGenerator`?

File: compiler/frontend/src/org/jetbrains/kotlin/resolve/AnnotationUtil.kt [hunk 2]
```diff
@@ -22,13 +22,11 @@ import org.jetbrains.kotlin.descriptors.PropertyAccessorDescriptor
 import org.jetbrains.kotlin.descriptors.annotations.AnnotationDescriptor
 import org.jetbrains.kotlin.descriptors.annotations.AnnotationUseSiteTarget
 import org.jetbrains.kotlin.descriptors.annotations.Annotations
-import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.resolve.DescriptorUtils
 
 public fun DeclarationDescriptor.hasPlatformStaticAnnotation(): Boolean {
```
Comment (udalov) [inline compiler/frontend/src/org/jetbrains/kotlin/resolve/AnnotationUtil.kt]: `hasJvmStaticAnnotation`?

**Comment by udalov** [comment]
Done

**Comment by yole** [comment]
Merged manually.

**Comment by udalov** [comment]
Would make sense to also rename other stuff in our project which has `PlatformStatic` in its name, e.g. `PlatformStaticAnnotationChecker`, `isPlatformStaticIn`, etc.

**Pull request closed without merging**